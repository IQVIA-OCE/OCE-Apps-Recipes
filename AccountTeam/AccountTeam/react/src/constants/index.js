import { Platform } from 'react-native';

export { SORT_OPTION_LIST, ACCESS_LEVEL_PICKLIST } from './sortOptionsList';
export { NAMESPACE, LOCALIZATION_NAMESPACE } from './namespacePrefix';
export { LOADING_STATUS } from './loadingStatus';
export const isIphone = !Platform.isPad && Platform.OS !== 'web';
