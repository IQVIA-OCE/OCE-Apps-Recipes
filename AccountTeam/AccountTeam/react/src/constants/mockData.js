import { NAMESPACE } from ".";
const customSettingsForTrue = [{ [`${NAMESPACE}DoNotUseTerritories255Field__c`]: true }];
const customSettingsForFalse = [{ [`${NAMESPACE}DoNotUseTerritories255Field__c`]: false }];
const territoryIds = [{ Territory2Id: '0MI4x0000000TLGGA2' }, { Territory2Id: '0MI4x0000000TLHGA2' }, { Territory2Id: '0MI4x0000000TLIGA2' }]
const territoryNameIds = [{ Id: '0MI4x0000000TLGGA2' }, { Id: '0MI4x0000000TLHGA2' }, { Id: '0MI4x0000000TLIGA2' }]
const territoryNames = [{ [`${NAMESPACE}Territories__c`]: ";DM - San Francisco 20D02;TM - SPC - Aurora 20A02T06;TM - SPC - Joliet 20A02T11;" }]
const teamMembers = [{
    Territory2: {
        AccountAccessLevel: "Read",
        Description: null,
        Name: "DM - San Francisco 20D02",
    },
    User: {
        City: null,
        Country: "US",
        Email: "test1@test.com",
        FirstName: "Partner",
        LastName: "Admin",
        Manager: null,
        Name: "Partner Admin",
        [`${NAMESPACE}LastSyncTime__c`]: null,
        Phone: null
    }
},
{
    Territory2: {
        AccountAccessLevel: "Edit",
        Description: null,
        Name: "TM - SPC - Aurora 20A02T06",
    },
    User: {
        City: null,
        Country: null,
        Email: "test2@test.com",
        FirstName: "Test",
        LastName: "User",
        Manager: null,
        Name: "Test User",
        [`${NAMESPACE}LastSyncTime__c`]: '2021-11-16T12:59:07.000+0000',
        Phone: "480"
    }
},
{
    Territory2: {
        AccountAccessLevel: "Edit",
        Description: null,
        Name: "TM - SPC - Aurora 20A02T06"
    },
    User: {
        City: "Chandler",
        Country: "United States",
        Email: "test3@test.com",
        FirstName: "test user 2",
        LastName: "user 2",
        Manager: null,
        Name: "test user 2",
        [`${NAMESPACE}LastSyncTime__c`]: null,
        Phone: "(480) 479-9000"
    }
}]
const searchTeamMemeberData = [
    {
        Territory2: {
            AccountAccessLevel: "Edit",
            Description: null,
            Name: "TM - SPC - Aurora 20A02T06",
        },
        User: {
            City: null,
            Country: null,
            Email: "test4@test.com",
            FirstName: "test user 1",
            LastName: "user 1",
            Manager: null,
            Name: "test user 1",
            [`${NAMESPACE}LastSyncTime__c`]: '2021-11-16T12:59:07.000+0000',
            Phone: "480"
        }
    },
    {
        Territory2: {
            AccountAccessLevel: "Edit",
            Description: null,
            Name: "TM - SPC - Aurora 20A02T06"
        },
        User: {
            City: "Chandler",
            Country: "United States",
            Email: "test4@test.com",
            FirstName: "test user 2",
            LastName: "user 2",
            Manager: null,
            Name: "test user 2",
            [`${NAMESPACE}LastSyncTime__c`]: null,
            Phone: "(480) 479-9000"
        }
    }];
const layoutData = {
    compactLayoutable: true,
    createable: true,
    custom: false,
    customSetting: false,
    deepCloneable: false,
    deletable: true,
    deprecatedAndHidden: false,
    extendedBy: null,
    extendsInterfaces: null,
    fields: [
        {
            label: "Account Access Level",
            length: 40,
            mask: null,
            maskType: null,
            name: "AccountAccessLevel",
            nameField: false,
            namePointing: false,
            nillable: false,
            permissionable: false,
            picklistValues: [
                {
                    active: true,
                    defaultValue: false,
                    label: "Read Only",
                    validFor: null,
                    value: "Read",
                },
                {
                    active: true,
                    defaultValue: false,
                    label: "Read/Write",
                    validFor: null,
                    value: "Edit",
                },
                {
                    active: true,
                    defaultValue: false,
                    label: "Owner",
                    validFor: null,
                    value: "All",
                },
            ],
        }
    ],
    implementsInterfaces: null,
    feedEnabled: false,
    queryable: true,
    defaultImplementation: null,
    labelPlural: "Meeting Members",
    networkScopeFieldName: null
};
export { territoryIds, customSettingsForTrue, customSettingsForFalse, territoryNames, teamMembers, searchTeamMemeberData, layoutData, territoryNameIds };