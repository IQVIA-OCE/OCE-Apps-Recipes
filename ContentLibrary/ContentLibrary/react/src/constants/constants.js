export const OWNED_BY_ME = 'ownedByMe';
export const SHARED_WITH_ME = 'sharedWithMe';
export const NO_CONNECTION = 'No Connection';