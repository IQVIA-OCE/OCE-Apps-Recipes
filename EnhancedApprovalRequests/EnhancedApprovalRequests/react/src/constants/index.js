export { LOADING_STATUS } from './loadingStatus';
export { APPROVAL_REQUEST_STATUS } from './approvalRequestStatus';
export * from './sorting';
