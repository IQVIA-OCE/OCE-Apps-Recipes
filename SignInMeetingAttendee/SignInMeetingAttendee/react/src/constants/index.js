import { Platform } from 'react-native';

export { NAMESPACE, userId, profileId, organizationId } from './namespacePrefix';
export const isIphone = !Platform.isPad;