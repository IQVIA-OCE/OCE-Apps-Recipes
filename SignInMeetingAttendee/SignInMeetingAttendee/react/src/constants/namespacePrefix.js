import { environment } from 'oce-apps-bridges';

export const NAMESPACE = environment.namespace();

export const userId = environment.userID();

export const profileId = environment.profileId();

export const organizationId = environment.organizationId();
