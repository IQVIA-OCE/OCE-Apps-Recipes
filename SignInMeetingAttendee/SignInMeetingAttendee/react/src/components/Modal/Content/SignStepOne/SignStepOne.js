import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { TextInput } from 'apollo-react-native';
import { useSelector, useDispatch } from 'react-redux';
import {
    updateFirstName, updateComments
} from '../../../../stores/meeting';
import { NAMESPACE } from '../../../../constants';
import { localized } from 'oce-apps-bridges';


const SignStepOne = () => {
    const { meetingDetails } = useSelector((state) => state.meeting);
    const dispatch = useDispatch();
    const onChangeName = val => {
        dispatch(updateFirstName(val))
    }
    const onChangeComment = val => {
        dispatch(updateComments(val))
    }
    return (
        <View style={styles.row}>
            <View style={styles.col}>
                <View style={styles.divider}>
                    <Text>{localized('Record_Type', 'Record Type')}</Text>
                    <Text style={styles.textDivider}>{meetingDetails[0]['RecordType.Name'] ? meetingDetails[0]['RecordType.Name'] : ''}</Text>
                </View>
                <View style={styles.divider}>
                    <Text>{localized('Meeting', 'Meeting')}<Text style={styles.mandatoryIcon}>*</Text></Text>
                    <Text style={styles.textDivider}>{meetingDetails[0].Name ? meetingDetails[0].Name : ''}</Text>
                </View>
                <View style={styles.divider}>
                    <TextInput
                        placeholder="Full Name"
                        label="Full Name"
                        required
                        style={styles.input}
                        onChangeText={onChangeName}
                    />
                </View>
            </View>
            <View style={styles.col}>
                <View style={styles.divider}>
                    <Text>{localized('Address', 'Address')}</Text>
                    <Text style={styles.textDivider}>{meetingDetails[0][`${NAMESPACE}Location__c`] ? meetingDetails[0][`${NAMESPACE}Location__c`] : ''}</Text>
                </View>
                <View style={styles.divider}>
                    <TextInput
                        label={localized('Global_Comments', 'Comments')}
                        multiline
                        rows={10}
                        minHeight={10}
                        style={styles.input}
                        onChangeText={onChangeComment}
                    />
                </View>
            </View>
        </View>
    )
};

export default SignStepOne;

const styles = StyleSheet.create({
    row: {
        flexDirection: 'row',
    },
    mandatoryIcon: {
        color: 'red',
    },
    input: {
        width: '95%',
    },
    divider: {
        marginTop: 30,
    },
    textDivider: {
        marginTop: 5,
    },
    col: {
        flex: 0.5
    },
});
