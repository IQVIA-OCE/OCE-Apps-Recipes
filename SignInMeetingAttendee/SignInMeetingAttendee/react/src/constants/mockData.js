import { NAMESPACE } from '../constants';

import { localized } from 'oce-apps-bridges';

const meetingLevelConfigRecord = [{
    [`${NAMESPACE}MealFieldType__c`]: "Picklist"
}];
const meetingLevelConfigRecordForcheck = [{
    [`${NAMESPACE}MealFieldType__c`]: "Check"
}];
const meetingMemberLayout = {
    mergeable: false,
    recordTypeInfos: [
        {
            urls: {
                layout: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/layouts/0129D000000vAQNQA2`
            },
            master: false,
            defaultRecordTypeMapping: true,
            available: true,
            developerName: "Attendee",
            active: true,
            recordTypeId: "0129D000000vAQNQA2",
            name: "Attendee"
        },
        {
            urls: {
                layout: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/layouts/0129D000000vAQOQA2`
            },
            master: false,
            defaultRecordTypeMapping: false,
            available: true,
            developerName: "Colleague",
            active: true,
            recordTypeId: "0129D000000vAQOQA2",
            name: "Colleague"
        }
    ],
    customSetting: false,
    isInterface: false,
    custom: true,
    label: "Meeting Member",
    replicateable: true,
    supportedScopes: [
        {
            name: "everything",
            label: "All meeting members"
        }
    ],
    actionOverrides: [
        {
            isAvailableInTouch: false,
            formFactor: "LARGE",
            url: null,
            pageId: "0M09D0000009clxSAA",
            name: "View"
        }
    ],
    compactLayoutable: true,
    associateParentEntity: null,
    extendsInterfaces: null,
    updateable: true,
    sobjectDescribeOption: "FULL",
    extendedBy: null,
    retrieveable: true,
    hasSubtypes: false,
    listviewable: null,
    namedLayoutInfos: [],
    implementedBy: null,
    isSubtype: false,
    associateEntityType: null,
    keyPrefix: "a42",
    undeletable: true,
    name: `${NAMESPACE}MeetingMember__c`,
    lookupLayoutable: null,
    searchLayoutable: false,
    deepCloneable: false,
    layoutable: true,
    deprecatedAndHidden: false,
    mruEnabled: false,
    createable: true,
    triggerable: true,
    fields: [
        {
            cascadeDelete: false,
            mask: null,
            createable: true,
            picklistValues: [
                {
                    defaultValue: false,
                    label: "Breakfast",
                    active: true,
                    value: "Breakfast",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Lunch",
                    active: true,
                    value: "Lunch",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Snack",
                    active: true,
                    value: "Snack",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Dinner",
                    active: true,
                    value: "Dinner",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Lunch & Snack",
                    active: true,
                    value: "Lunch & Snack",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Did Not Consume",
                    active: true,
                    value: "Did Not Consume",
                    validFor: null
                },
                {
                    defaultValue: false,
                    label: "Not Applicable",
                    active: true,
                    value: "Not Applicable",
                    validFor: null
                }
            ],
            referenceTo: [],
            formulaTreatNullNumberAsZero: false,
            relationshipName: null,
            restrictedDelete: false,
            soapType: "xsd:string",
            idLookup: false,
            defaultValue: null,
            inlineHelpText: null,
            maskType: null,
            custom: true,
            autoNumber: false,
            defaultedOnCreate: false,
            aggregatable: true,
            controllerName: null,
            type: "picklist",
            dependentPicklist: false,
            referenceTargetField: null,
            htmlFormatted: false,
            namePointing: false,
            filterable: true,
            aiPredictionField: false,
            digits: 0,
            length: 255,
            calculated: false,
            deprecatedAndHidden: false,
            byteLength: 765,
            precision: 0,
            name: `${NAMESPACE}MealOption__c`,
            filteredLookupInfo: null,
            displayLocationInDecimal: false,
            scale: 0,
            groupable: true,
            defaultValueFormula: null,
            compoundFieldName: null,
            polymorphicForeignKey: false,
            nillable: true,
            encrypted: false,
            restrictedPicklist: false,
            highScaleNumber: false,
            sortable: true,
            nameField: false,
            permissionable: true,
            writeRequiresMasterRead: false,
            queryByDistance: false,
            unique: false,
            extraTypeInfo: null,
            externalId: false,
            searchPrefilterable: false,
            caseSensitive: false,
            relationshipOrder: null,
            updateable: true,
            calculatedFormula: null,
            label: "Meal Option"
        },
        {
            cascadeDelete: false,
            mask: null,
            createable: true,
            picklistValues: [],
            referenceTo: [],
            formulaTreatNullNumberAsZero: false,
            relationshipName: null,
            restrictedDelete: false,
            soapType: "xsd:boolean",
            idLookup: false,
            defaultValue: false,
            inlineHelpText: "Determines if the Meeting Member requires a meal at the meeting.",
            maskType: null,
            custom: true,
            autoNumber: false,
            defaultedOnCreate: true,
            aggregatable: false,
            controllerName: null,
            type: "boolean",
            dependentPicklist: false,
            referenceTargetField: null,
            htmlFormatted: false,
            namePointing: false,
            filterable: true,
            aiPredictionField: false,
            digits: 0,
            length: 0,
            calculated: false,
            deprecatedAndHidden: false,
            byteLength: 0,
            precision: 0,
            name: `${NAMESPACE}Meal__c`,
            filteredLookupInfo: null,
            displayLocationInDecimal: false,
            scale: 0,
            groupable: true,
            defaultValueFormula: null,
            compoundFieldName: null,
            polymorphicForeignKey: false,
            nillable: false,
            encrypted: false,
            restrictedPicklist: false,
            highScaleNumber: false,
            sortable: true,
            nameField: false,
            permissionable: true,
            writeRequiresMasterRead: false,
            queryByDistance: false,
            unique: false,
            extraTypeInfo: null,
            externalId: false,
            searchPrefilterable: false,
            caseSensitive: false,
            relationshipOrder: null,
            updateable: true,
            calculatedFormula: null,
            label: "Meal?"
        }
    ],
    deletable: true,
    activateable: false,
    searchable: false,
    childRelationships: [
        {
            junctionReferenceTo: [],
            deprecatedAndHidden: false,
            childSObject: "AttachedContentDocument",
            restrictedDelete: false,
            relationshipName: "AttachedContentDocuments",
            cascadeDelete: true,
            field: "LinkedEntityId",
            junctionIdListNames: []
        }
    ],
    urls: {
        layouts: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/layouts`,
        quickActions: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/quickActions`,
        sobject: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c`,
        uiNewRecord: `https://oce-pipeline-2--dcdm201002.my.salesforce.com/a42/e`,
        describe: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe`,
        compactLayouts: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/compactLayouts`,
        approvalLayouts: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/describe/approvalLayouts`,
        uiDetailTemplate: "https://oce-pipeline-2--dcdm201002.my.salesforce.com/{ID}",
        rowTemplate: `/services/data/v50.0/sobjects/${NAMESPACE}MeetingMember__c/{ID}`,
        uiEditTemplate: "https://oce-pipeline-2--dcdm201002.my.salesforce.com/{ID}/e"
    },
    implementsInterfaces: null,
    feedEnabled: false,
    queryable: true,
    defaultImplementation: null,
    labelPlural: "Meeting Members",
    networkScopeFieldName: null
};


const meetingsInfo = [{
    CreatedBy: {
        Name: "Sales Representative PHX2"
    },
    CreatedById: "0056F00000B45efQAB",
    CreatedDate: "2021-08-24T18:35:54.000+0000",
    Id: "a470k0000010AMgAAM",
    Name: "Test1",
    [`${NAMESPACE}EndDateTime__c`]: "2021-08-31T18:35:00.000+0000",
    [`${NAMESPACE}Location__c`]: null,
    [`${NAMESPACE}StartDateTime__c`]: "2021-08-30T18:35:00.000+0000",
    [`${NAMESPACE}Status__c`]: "Draft",
    [`${NAMESPACE}Topic__c`]: null,
    RecordType: {
        Name: "Speaker Meeting"
    },
    RecordTypeId: "0120p0000012JBgAAM",
    uid: "a470k0000010AMgAAM",

}];


const meetingsMemebersInfo = [{
    Id: "a420k0000005MvUAAU",
    Name: "ASHOK KUMAR",
    [`${NAMESPACE}Address__c`]: null,
    [`${NAMESPACE}AttendanceStatus__c`]: "Attended-eSigned",
    [`${NAMESPACE}AttendeeType__c`]: null,
    [`${NAMESPACE}DiscrepancyReason__c`]: "Exceeded Monetary Caps Limit Warning",
    [`${NAMESPACE}MealOption__c`]: "Dinner",
    [`${NAMESPACE}SignatureDate__c`]: "2021-08-27T10:55:43.000+0000",
    [`${NAMESPACE}SpecialtyText__c`]: "Family medicine",
    [`${NAMESPACE}Status__c`]: "Invited",
    RecordTypeId: "0120p0000012JBOAA2",
    uid: "a420k0000005MvUAAU",
}, {
    Id: "a420k0000005Mv5AAE",
    Name: "Sales Representative PHX2",
    [`${NAMESPACE}Address__c`]: null,
    [`${NAMESPACE}AttendanceStatus__c`]: "Attended-eSigned",
    [`${NAMESPACE}AttendeeType__c`]: null,
    [`${NAMESPACE}DiscrepancyReason__c`]: null,
    [`${NAMESPACE}MealOption__c`]: null,
    [`${NAMESPACE}SignatureDate__c`]: null,
    [`${NAMESPACE}SpecialtyText__c`]: null,
    [`${NAMESPACE}Status__c`]: "Accepted",
    RecordTypeId: "0120p0000012JBPAA2",
    uid: "a420k0000005Mv5AAE",
}];

const meetingConfigData = [{
    [`${NAMESPACE}AttendStatusesForTopicCompliance__c`]: null,
    [`${NAMESPACE}EnableConsent__c`]: 1,
    [`${NAMESPACE}EnableWriteIns__c`]: 1,
    [`${NAMESPACE}IsEnabledSignInDateValidation__c`]: 1,
    [`${NAMESPACE}SignInAttendanceStatuses__c`]: null,
    [`${NAMESPACE}SignInInvitationStatuses__c`]: null,
    [`${NAMESPACE}ValidateDaysInPastForTopicForWalkIn__c`]: null,
    [`${NAMESPACE}ValidateDaysInPastForTopic__c`]: null,
    [`${NAMESPACE}ValidateRestrictedProductsForWalkIns__c`]: 'warn',

}];

const records = [{
    CreatedBy: {
        Name: "Sales Representative PHX2"
    },
    CreatedById: "0056F00000B45efQAB",
    CreatedDate: "2021-08-24T18:35:54.000+0000",
    Id: "a470k0000010AMgAAM",
    Name: "Test1",
    [`${NAMESPACE}EndDateTime__c`]: "2021-10-13T18:35:00.000+0000",
    [`${NAMESPACE}Location__c`]: 'Test Location',
    [`${NAMESPACE}StartDateTime__c`]: "2021-10-13T18:35:00.000+0000",
    [`${NAMESPACE}Status__c`]: "Draft",
    [`${NAMESPACE}Topic__c`]: null,
    ['RecordType.Name']: "Speaker Meeting",
    RecordTypeId: "0120p0000012JBgAAM",
    uid: "a470k0000010AMgAAM",
    [`${NAMESPACE}MeetingMember__r`]: {
        records: meetingsMemebersInfo
    }


}];

const singleRecordRow = {
    Id: "a421s000000CuyfAAC",
    Name: "Test A1",
    [`${NAMESPACE}Address__c`]: null,
    [`${NAMESPACE}AttendanceStatus__c`]: "Attended-eSigned",
    [`${NAMESPACE}AttendeeType__c`]: null,
    [`${NAMESPACE}DiscrepancyReason__c`]: null,
    [`${NAMESPACE}MealOption__c`]: "Dinner",
    [`${NAMESPACE}SignatureDate__c`]: "2021-09-15T10:33:45.421+05:30",
    [`${NAMESPACE}SpecialtyText__c`]: "Addiction medicine",
    [`${NAMESPACE}Status__c`]: "Invited",
    RecordTypeId: "0120k000000GlNkAAK",
    key: "a421s000000CuyfAAC",
    uid: "a421s000000CuyfAAC",

}

const meetingHighLightPanel = [{
    fieldItems: [{
        label: localized('Name', 'Meeting Name'),
        layoutComponents: [{
            type: "Field",
            value: "Name",
            details: {
                name: "Name",
                type: "string"
            }
        }]
    },
    {
        label: `${localized('Start', 'Start')}${localized('Date_time', 'Date/Time')}`,
        layoutComponents: [{
            type: `Field`,
            value: `${NAMESPACE}StartDateTime__c`,
            details: {
                name: `${NAMESPACE}StartDateTime__c`,
                type: "datetime"
            }
        }],
    },
    {
        label: `${localized('InputDateRange_EndDate_DefaultLabel', 'End')}${localized('Date_time', 'Date/Time')}`,
        layoutComponents: [{
            type: "Field",
            value: `${NAMESPACE}EndDateTime__c`,
            details: {
                name: `${NAMESPACE}EndDateTime__c`,
                type: "datetime"
            }
        }],
    },
    {
        label: `${localized('AttachmentCreatedByIdLabel', 'Created By')}`,
        layoutComponents: [
            {
                type: "Field",
                value: "CreatedById",
                details: {
                    name: "CreatedBy",
                    type: "reference",
                    relationshipName: 'CreatedBy'
                }
            },
            {
                type: "Separator",
                value: ", ",
            },
            {
                type: "Field",
                value: "CreatedDate",
                details: {
                    name: "CreatedDate",
                    type: "datetime"
                }
            },
        ],
    },
    {
        label: `${localized('Location', 'Location')}`,
        layoutComponents: [{
            type: "Field",
            value: `${NAMESPACE}Location__c`,
            details: {
                name: `${NAMESPACE}Location__c`,
                type: "string"
            }
        }],
    }]
}]

const meetingHighLight = [{
    actions: null,
    fieldItems: [{
        editableForNew: true,
        editableForUpdate: true,
        label: "Meeting Name",
        layoutComponents: [{
            displayLines: 1,
            tabOrder: 1,
            type: "Field",
            value: "Name",
            details: {
                label: "Meeting Name",
                length: 80,
                mask: null,
                maskType: null,
                name: "Name",
                type: "string",
                unique: false,
                updateable: true,
                writeRequiresMasterRead: false
            }
        }],
        placeholder: false,
        required: false,
    }, {
        editableForNew: true,
        editableForUpdate: true,
        label: "Start Date/Time",
        layoutComponents: [{
            displayLines: 1,
            tabOrder: 2,
            type: "Field",
            value: [`${NAMESPACE}StartDateTime__c`],
            details: {
                label: "Start Date/Time",
                length: 0,
                mask: null,
                maskType: null,
                name: [`${NAMESPACE}StartDateTime__c`],
                type: "datetime"
            }
        }],
        placeholder: false,
        required: false,
    }],
    id: "0AH0k000000CspjGAC",
    imageItems: null,
    label: "Summary",
    name: [`${NAMESPACE}Summary`],
    objectType: [`${NAMESPACE}Meeting__c`],
}]


export {
    records, meetingLevelConfigRecord, meetingMemberLayout,
    meetingsInfo, meetingsMemebersInfo, meetingConfigData,
    singleRecordRow, meetingHighLight, meetingHighLightPanel, meetingLevelConfigRecordForcheck
};

