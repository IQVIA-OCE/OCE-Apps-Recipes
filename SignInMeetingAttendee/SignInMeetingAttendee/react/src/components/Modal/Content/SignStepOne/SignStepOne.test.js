import React from 'react';
import SignStepOne from './SignStepOne';
import renderer from 'react-test-renderer';
import { useSelector, useDispatch } from 'react-redux';
import { records } from "../../../../constants/mockData";


jest.useFakeTimers();


jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("../../../../../bridge/Localization/localization.native");

describe('Signing step one', () => {
    it('SigningStepOne Component should render properly', () => {
        useDispatch.mockReturnValue(() => jest.fn());
        useSelector.mockImplementation((cb) => cb({
            meeting: {
                meetingDetails: records,
                signature: null
            }
        }));
        const tree = renderer.create(
            <SignStepOne />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

});
