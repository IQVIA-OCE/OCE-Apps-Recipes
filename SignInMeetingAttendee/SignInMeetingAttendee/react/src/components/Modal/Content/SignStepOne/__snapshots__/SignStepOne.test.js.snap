// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signing step one SigningStepOne Component should render properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "marginTop": 5,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <Text>
        <Text
          style={
            Object {
              "color": "red",
            }
          }
        >
          *
        </Text>
      </Text>
      <Text
        style={
          Object {
            "marginTop": 5,
          }
        }
      >
        Test1
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": 176,
            },
            Object {
              "width": "95%",
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 8,
            }
          }
        >
          Full Name
          <Text
            style={
              Object {
                "color": "#E20000",
              }
            }
          >
             *
          </Text>
        </Text>
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {
                "width": "95%",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              paddingLeft={8}
              paddingRight={8}
              placeholder="Full Name"
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "marginTop": 5,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": 176,
            },
            Object {
              "width": "95%",
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {
                "width": "95%",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 200,
                "minHeight": 200,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              maxResizableWidth={750}
              minHeight={10}
              minWidth={0}
              multiline={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              paddingLeft={8}
              paddingRight={8}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 200,
                    "minHeight": 0,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
